test/honeybadger_test.exs

```diff
@@ -9,6 +9,108 @@ defmodule HoneybadgerTest do
     on_exit(&Honeybadger.API.stop/0)
   end
 
+  describe "exclude_error option" do
+    test "by default, server gets notified of errors" do
+      restart_with_config(exclude_envs: [])
+
+      fun = fn :num -> nil end
+
+      try do
+        fun.(:boom)
+      rescue
+        exception ->
+          :ok = Honeybadger.notify(exception, stacktrace: __STACKTRACE__)
+      end
+
+      assert_receive {:api_request, %{"error" => error}}
+      assert error["class"] == "FunctionClauseError"
+    end
+
+    test "excludes errors sent to server when exclude_error?/1 returns true" do
+      defmodule ExcludeFunClauseErrors do
+        alias Honeybadger.ExcludeErrors
+
+        @behaviour ExcludeErrors
+
+        @impl ExcludeErrors
+        def exclude_error?(notice) do
+          notice.error.class == "FunctionClauseError"
+        end
+      end
+
+      restart_with_config(exclude_envs: [], exclude_errors: ExcludeFunClauseErrors)
+
+      fun = fn :num -> nil end
+
+      try do
+        fun.(:boom)
+      rescue
+        exception ->
+          nil = Honeybadger.notify(exception, stacktrace: __STACKTRACE__)
+      end
+
+      refute_receive {:api_request, _}
+    end
+
+    test "errors are sent to server when exclude_error?/1 returns false" do
+      defmodule ExcludeErrors do
+        alias Honeybadger.ExcludeErrors
+
+        @behaviour ExcludeErrors
+
+        @impl ExcludeErrors
+        def exclude_error?(notice) do
+          notice.error.class == "RuntimeError"
+        end
+      end
+
+      restart_with_config(exclude_envs: [], exclude_errors: ExcludeErrors)
+
+      fun = fn :num -> nil end
+
+      try do
+        fun.(:boom)
+      rescue
+        exception ->
+          :ok = Honeybadger.notify(exception, stacktrace: __STACKTRACE__)
+      end
+
+      assert_receive {:api_request, %{"error" => error}}
+      assert error["class"] == "FunctionClauseError"
+    end
+
+    test "errors are sent to server when error is missing in list of errors passed" do
+      restart_with_config(exclude_envs: [], exclude_errors: ["RuntimeError"])
+
+      fun = fn :num -> nil end
+
+      try do
+        fun.(:boom)
+      rescue
+        exception ->
+          :ok = Honeybadger.notify(exception, stacktrace: __STACKTRACE__)
+      end
+
+      assert_receive {:api_request, %{"error" => error}}
+      assert error["class"] == "FunctionClauseError"
+    end
+
+    test "excludes errors sent to server when a list of errors is passed" do
+      restart_with_config(exclude_envs: [], exclude_errors: ["FunctionClauseError"])
```

Comment:
Is it possible to also support error classes themselves in addition to the string `error.name`? I think some people will configure `exclude_errors: [FunctionClauseError]`.

Reply:
Yes, this would be possible . We might need to convert the exception classes to string first then check against the error name. 
```elixir
Enum.map([FunctionClauseError], & &1|> to_string() |> String.trim_leading("Elixir."))
```